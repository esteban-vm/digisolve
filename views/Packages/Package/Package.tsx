import type { PackageComponent } from '@/types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCheck } from '@fortawesome/free-solid-svg-icons'
import { Button } from '@/components'
import { formatPrice } from '@/helpers'
import { styled } from '@/styles'

const PackageComponent: PackageComponent = ({ heading, price, duration, benefits, ...rest }) => {
  const formattedPrice = formatPrice(price)

  return (
    <article {...rest}>
      <div className='package'>
        <div className='package__section'>
          <h4 className='package__heading'>{heading}</h4>
          <p className='package__paragraph'>
            {formattedPrice} <span>{duration} months</span>
          </p>
          <p className='package__paragraph'>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Modi, sit nemo natus ad vero molestias?
          </p>
        </div>
        <div className='package__section'>
          <ul className='package__benefit--list'>
            {benefits.map((benefit, index) => (
              <li key={index} className='package__benefit--item'>
                <FontAwesomeIcon icon={faCheck} className='package__benefit--icon' />
                {benefit}
              </li>
            ))}
          </ul>
        </div>
        <div className='package__section'>
          <Button text='Sign up now' isFull />
        </div>
      </div>
    </article>
  )
}

const Package = styled(PackageComponent)`
  .package {
    background-color: var(--color-white);
    width: 90%;
    margin: 0 auto 30px;
    box-shadow: 0 3px 2px var(--color-shadow);
    text-align: left;

    &__section {
      padding: 15px;
      border-bottom: 1px solid var(--color-shadow);

      :first-of-type {
        background-color: var(--color-lighter);
      }

      :last-of-type {
        border: 0;
        text-align: center;

        span {
          margin: 15px 0;
        }
      }
    }

    &__heading {
      font-size: 150%;
      color: var(--color-grey);
      font-weight: 400;
      margin-bottom: 50px;
      text-transform: capitalize;
    }

    &__paragraph {
      :first-of-type {
        color: var(--color-primary);
        font-size: 300%;
        font-weight: 300;
        margin-bottom: 10px;

        span {
          font-size: 30%;
          font-weight: 500;
        }
      }

      :last-of-type {
        color: var(--color-grey);
        font-size: 80%;
        margin-bottom: 10px;
      }
    }

    &__benefit {
      &--list {
        list-style: none;
      }

      &--item {
        color: var(--color-grey);
        padding: 5px 0;
      }

      &--icon {
        width: 1.2rem;
        height: 1.2rem;
        display: inline-block;
        margin-right: 10px;
        color: var(--color-primary);
      }
    }
  }
`

export default Package
