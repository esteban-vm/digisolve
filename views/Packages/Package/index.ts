export { default, type PackageProps } from './Package'
