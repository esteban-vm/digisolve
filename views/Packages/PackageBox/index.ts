export { default, type PackageProps } from './PackageBox'
