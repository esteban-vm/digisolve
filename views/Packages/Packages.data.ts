import type { PackageProps } from './Package'

const description = 'Ratione similique aut incidunt repellendus excepturi ut commodi laboriosam.'

const packages: PackageProps[] = [
  {
    id: 'd7e645e0-e825-11ed-b24d-893619994521',
    title: 'starter',
    price: 699,
    duration: 3,
    description,
    features: [
      'SEO, SMO and PPC',
      '2 Blogs per month',
      'Facebook & Twitter Marketing',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
  {
    id: 'df095c40-e825-11ed-b24d-893619994521',
    title: 'corporate',
    price: 2999,
    duration: 6,
    description,
    features: [
      'SEO, SMO and PPC',
      '4 Blogs per month',
      'All Social Media Channels',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
  {
    id: 'e21563c0-e825-11ed-b24d-893619994521',
    title: 'enterprice',
    price: 4999,
    duration: 6,
    description,
    features: [
      'Personalized Marketing',
      '8 Blogs per month',
      'All Social Media Channels',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
]

export default packages
