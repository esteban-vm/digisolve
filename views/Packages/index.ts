export { default as Packages } from './Packages'
