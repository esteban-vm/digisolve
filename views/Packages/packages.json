[
  {
    "title": "starter",
    "price": 699,
    "duration": 3,
    "features": [
      "SEO, SMO and PPC",
      "2 Blogs per month",
      "Facebook & Twitter Marketing",
      "Monthly Performance",
      "Traffic Report"
    ]
  },
  {
    "title": "corporate",
    "price": 2999,
    "duration": 6,
    "features": [
      "SEO, SMO and PPC",
      "4 Blogs per month",
      "All Social Media Channels",
      "Monthly Performance",
      "Traffic Report"
    ]
  },
  {
    "title": "enterprice",
    "price": 4999,
    "duration": 6,
    "features": [
      "Personalized Marketing",
      "8 Blogs per month",
      "All Social Media Channels",
      "Monthly Performance",
      "Traffic Report"
    ]
  }
]
