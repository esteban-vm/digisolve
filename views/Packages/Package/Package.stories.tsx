import type { Meta, StoryObj } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import { packages } from '../Packages'
import Package from './Package'

const Wrapper = styled.div`
  width: 75vw;
`

export default {
  argTypes: {
    benefits: {
      control: false,
    },
  },
  component: Package,
  title: 'Views/Packages/Packages',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
            <Grid.Col isOneThird>
              <Story {...context} />
            </Grid.Col>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
} as Meta<typeof Package>

const Examples: StoryObj<typeof Package>[] = []
for (const pack of packages) Examples.push({ args: pack })

export const Example1 = Examples[0]
export const Example2 = Examples[1]
export const Example3 = Examples[2]
