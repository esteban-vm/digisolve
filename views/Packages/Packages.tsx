import type { Component, PackageComponentProps } from '@/types'
import { useId } from 'react'
import { Grid } from '@/components'
import { styled } from '@/styles'
import Package from './Package'

export const packages: PackageComponentProps[] = [
  {
    heading: 'starter',
    price: 699,
    duration: 3,
    benefits: [
      'SEO, SMO and PPC',
      '2 Blogs per month',
      'Facebook & Twitter Marketing',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
  {
    heading: 'corporate',
    price: 2999,
    duration: 6,
    benefits: [
      'SEO, SMO and PPC',
      '4 Blogs per month',
      'All Social Media Channels',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
  {
    heading: 'enterprice',
    price: 4999,
    duration: 6,
    benefits: [
      'Personalized Marketing',
      '8 Blogs per month',
      'All Social Media Channels',
      'Monthly Performance',
      'Traffic Report',
    ],
  },
]

const PackagesComponent: Component = (props) => {
  const id = useId()

  return (
    <section aria-labelledby={id} {...props}>
      <Grid.Row>
        <h2 id={id} className='packages__heading'>
          Check out our most popular packages
        </h2>
      </Grid.Row>

      <Grid.Row>
        {packages.map((pack, index) => (
          <Grid.Col key={index} isOneThird>
            <Package {...pack} />
          </Grid.Col>
        ))}
      </Grid.Row>
    </section>
  )
}

const Packages = styled(PackagesComponent)`
  text-align: center;
  background-color: var(--color-light);

  .packages__heading {
    color: var(--color-black);
    font-size: 250%;
    margin-bottom: 20px;
    margin-top: 150px;
    text-transform: capitalize;
  }
`

export default Packages
