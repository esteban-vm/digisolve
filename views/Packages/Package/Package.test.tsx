import { render, cleanup, screen, create } from '@/tests'
import { formatPrice } from '@/helpers'
import { packages } from '../Packages'
import Package from './Package'

describe('ðŸ§ª PACKAGE:', () => {
  const [testPackage] = packages

  describe('display tests:', () => {
    let parent: HTMLElement

    beforeEach(() => {
      render(<Package {...testPackage} />)
      parent = screen.getByRole('article')
    })

    afterEach(cleanup)

    it('should display the heading', () => {
      const heading = screen.getByRole('heading', { name: testPackage.heading, level: 4 })
      expect(heading).toBeInTheDocument()
      expect(heading).toBeVisible()
      expect(parent).toContainElement(heading)
    })

    it('should display the price', () => {
      const price = screen.getByText(formatPrice(testPackage.price))
      expect(price).toBeInTheDocument()
      expect(price).toBeVisible()
      expect(parent).toContainElement(price)
    })

    it('should display the duration', () => {
      const duration = screen.getByText(`${testPackage.duration} months`)
      expect(duration).toBeInTheDocument()
      expect(duration).toBeVisible()
      expect(parent).toContainElement(duration)
    })

    it('should display the description', () => {
      const description = screen.getByText(/^lorem ipsum/i)
      expect(description).toBeInTheDocument()
      expect(description).toBeVisible()
      expect(parent).toContainElement(description)
    })

    it('should display the benefits', () => {
      const list = screen.getByRole('list')
      const benefits = screen.getAllByRole('listitem')
      const icons = screen.getAllByRole('img', { hidden: true })
      expect(icons.length).toBe(benefits.length)

      for (let index = 0; index < benefits.length; index++) {
        const benefit = benefits[index]
        expect(benefit).toBeInTheDocument()
        expect(benefit).toBeVisible()
        expect(benefit).toContainElement(icons[index])
        expect(benefit).toHaveTextContent(testPackage.benefits[index])
        expect(list).toContainElement(benefit)
      }

      expect(parent).toContainElement(list)
    })

    it('should display the button', () => {
      const button = screen.getByRole('button', { name: /^sign up now$/i })
      expect(button).toBeInTheDocument()
      expect(button).toBeVisible()
      expect(parent).toContainElement(button)
    })
  })

  describe('style tests', () => {
    it('should render with correct styles', () => {
      const tree = create(<Package {...testPackage} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
