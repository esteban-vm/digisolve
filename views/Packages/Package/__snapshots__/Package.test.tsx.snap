// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª PACKAGE: style tests should render with correct styles 1`] = `
.emotion-0 .package {
  background-color: var(--color-white);
  width: 90%;
  margin: 0 auto 30px;
  box-shadow: 0 3px 2px var(--color-shadow);
  text-align: left;
}

.emotion-0 .package__section {
  padding: 15px;
  border-bottom: 1px solid var(--color-shadow);
}

.emotion-0 .package__section:first-of-type {
  background-color: var(--color-lighter);
}

.emotion-0 .package__section:last-of-type {
  border: 0;
  text-align: center;
}

.emotion-0 .package__section:last-of-type span {
  margin: 15px 0;
}

.emotion-0 .package__heading {
  font-size: 150%;
  color: var(--color-grey);
  font-weight: 400;
  margin-bottom: 50px;
  text-transform: capitalize;
}

.emotion-0 .package__paragraph:first-of-type {
  color: var(--color-primary);
  font-size: 300%;
  font-weight: 300;
  margin-bottom: 10px;
}

.emotion-0 .package__paragraph:first-of-type span {
  font-size: 30%;
  font-weight: 500;
}

.emotion-0 .package__paragraph:last-of-type {
  color: var(--color-grey);
  font-size: 80%;
  margin-bottom: 10px;
}

.emotion-0 .package__benefit--list {
  list-style: none;
}

.emotion-0 .package__benefit--item {
  color: var(--color-grey);
  padding: 5px 0;
}

.emotion-0 .package__benefit--icon {
  width: 1.2rem;
  height: 1.2rem;
  display: inline-block;
  margin-right: 10px;
  color: var(--color-primary);
}

.emotion-1 {
  display: inline-block;
}

.emotion-1 .btn {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-1 .btn:hover,
.emotion-1 .btn:active {
  background-color: var(--color-dark-primary);
}

<article
  className="emotion-0"
>
  <div
    className="package"
  >
    <div
      className="package__section"
    >
      <h4
        className="package__heading"
      >
        starter
      </h4>
      <p
        className="package__paragraph"
      >
        $699
         
        <span>
          3
           months
        </span>
      </p>
      <p
        className="package__paragraph"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Modi, sit nemo natus ad vero molestias?
      </p>
    </div>
    <div
      className="package__section"
    >
      <ul
        className="package__benefit--list"
      >
        <li
          className="package__benefit--item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check package__benefit--icon"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          SEO, SMO and PPC
        </li>
        <li
          className="package__benefit--item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check package__benefit--icon"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          2 Blogs per month
        </li>
        <li
          className="package__benefit--item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check package__benefit--icon"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Facebook & Twitter Marketing
        </li>
        <li
          className="package__benefit--item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check package__benefit--icon"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Monthly Performance
        </li>
        <li
          className="package__benefit--item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check package__benefit--icon"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Traffic Report
        </li>
      </ul>
    </div>
    <div
      className="package__section"
    >
      <span
        className="emotion-1"
      >
        <a
          className="btn"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          role="button"
        >
          Sign up now
        </a>
      </span>
    </div>
  </div>
</article>
`;
